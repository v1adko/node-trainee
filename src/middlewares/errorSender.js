import errorsResponses from '../utils/errorsResponses';

const errorSender = (error, requset, response, next) => {
  const errorResponse = errorsResponses[error.name];
  if (errorResponse) {
    response.status(error.status || 500).json(errorResponse);
  } else {
    response.status(error.status || 500).json(error.message);
  }

  next();
};

export default errorSender;
