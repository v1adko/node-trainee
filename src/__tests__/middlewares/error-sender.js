import errorSender from '../../middlewares/error-sender';

class ResponseMock {
  status(code) {
    this.code = code;
    return this;
  }

  json(responseObject) {
    this.responseObject = responseObject;
    return this;
  }
}

let counter = 0;

describe('Test error sender', () => {
  beforeEach(() => {
    counter = 0;
  });

  it('should send common respoce object', () => {
    const error = new Error('test error message');
    const responce = new ResponseMock();
    errorSender(error, null, responce, () => {
      counter += 1;
    });
    const { responseObject, code } = responce;

    expect(counter).toBe(1);
    expect(code).toBe(500);
    expect(responseObject).toMatchSnapshot();
  });
});
