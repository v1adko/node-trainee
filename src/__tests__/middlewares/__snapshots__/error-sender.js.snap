// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test error sender should send common respoce object 1`] = `
Object {
  "error": Object {
    "code": 500,
    "message": "Error(500): test error message",
    "name": "Error",
    "payload": undefined,
  },
}
`;
